central/graphql/resolvers/utils.go

```diff
@@ -357,3 +357,13 @@ func V1RawQueryAsResolverQuery(rQ *v1.RawQuery) (RawQuery, PaginatedQuery) {
 		},
 	}
 }
+
+// ErrorOnQueryContainingField logs error if the query contains the given field label
+func ErrorOnQueryContainingField(query *v1.Query, label search.FieldLabel, errMsg string) {
+	search.ApplyFnToAllBaseQueries(query, func(bq *v1.BaseQuery) {
+		mfQ, ok := bq.GetQuery().(*v1.BaseQuery_MatchFieldQuery)
+		if ok && mfQ.MatchFieldQuery.GetField() == label.String() {
+			log.Errorf(errMsg)
```

Comment:
Consider adding a suffix `Response maybe unexpected` to the incoming message. Since we are not erroring out, the execution continues on the caller side, and may lead to undesired results.

Reply:
My 2c:
`log.Errorf("Unexpected field (%s) found in query on subresolver (%s)", label.String(), resolver)`
Then all you have to pass in is the field to check for and the name of the subresolver we're checking in